semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/DenotationOps.scala

```diff
@@ -104,19 +105,46 @@ trait DenotationOps { self: DatabaseOps =>
       }
     }
 
+    private def newInfo: (Option[s.Type], List[g.Symbol]) = {
+      gsym.info.toSemantic
+    }
+
     private def overrides: List[m.Symbol] =
       if (config.overrides.isAll) gsym.overrides.map(_.toSemantic)
       else Nil
 
-    def toDenotation(saveOverrides: Boolean): m.Denotation = {
-      val (minfo, mnames) = info
-      val over = if (saveOverrides) overrides else Nil
-      m.Denotation(flags, name, minfo, mnames, Nil, over)
+    def toDenotation(saveOverrides: Boolean): DenotationResult = {
+      val over = {
+        if (saveOverrides) overrides
+        else Nil
+      }
+      val todoOverrides = {
+        if (saveOverrides && config.denotations.saveReferences) gsym.overrides
+        else Nil
+      }
+      config.signatures match {
+        case SignatureMode.None =>
+          val denot = m.Denotation(flags, name, "", Nil, Nil, over, None)
+          DenotationResult(denot, todoOverrides, Nil)
+        case SignatureMode.Old =>
+          val (signature, names) = oldInfo
+          val denot = m.Denotation(flags, name, signature, names, Nil, over, None)
+          DenotationResult(denot, todoOverrides, Nil)
+        case SignatureMode.New =>
+          val (tpe, todoTpe) = newInfo
+          val denot = m.Denotation(flags, name, "", Nil, Nil, over, tpe)
+          DenotationResult(denot, todoOverrides, todoTpe)
+        case SignatureMode.All =>
```

Comment:
Will there ever be a use-case for this mode?

Reply:
Transition period when tools like Metals still don't have support for s.Type prettyprinting.

Reply:
I feel like the transition will more likely be atomic, we throw out the pretty printing and add in s.Type.

Reply:
Let's keep this for now. It's not like supporting this mode costs us anything apart from single-digit amount of lines of code.